src/WebUI/WorkLoads/DeploymentsTable.tsx

```diff
@@ -255,8 +256,28 @@ export class DeploymentsTable extends BaseComponent<IDeploymentsTableProperties,
         return renderTableCell(rowIndex, columnIndex, tableColumn, itemToRender);
     }
 
-    private static _renderPodsCountCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<IDeploymentReplicaSetItem>, deployment: IDeploymentReplicaSetItem): JSX.Element {
-        return renderPodsStatusTableCell(rowIndex, columnIndex, tableColumn, deployment.pods, deployment.statusProps, deployment.podsTooltip);
+    private _renderPodsCountCell = (rowIndex: number, columnIndex: number, tableColumn: ITableColumn<IDeploymentReplicaSetItem>, deployment: IDeploymentReplicaSetItem): JSX.Element => {
+        const replicaSet = this._getSelectedReplicaSet(deployment);
+        const podslist = StoreManager.GetStore<PodsStore>(PodsStore).getState().podsList;
+        const relevantPods = (podslist && podslist.items && podslist.items.filter(p => (replicaSet && Utils.isOwnerMatched(p.metadata, replicaSet.metadata.uid)) || false))
+
+        const _onPodsClicked = (relevantPods && replicaSet) ? (() => {
+            const selectedPod = relevantPods.find(p => PodPhaseToStatus[p.status.phase] === Statuses.Failed) || relevantPods[0];
+            this._selectionActionCreator.selectItem({
+                item: selectedPod,
+                itemUID: selectedPod.metadata.uid,
+                selectedItemType: SelectedItemKeys.PodDetailsKey,
+                showSelectedItem: true,
+                properties: {
+                    pods: relevantPods,
+                    parentItemKind: replicaSet.kind || "ReplicaSet",
+                    parentItemName: replicaSet.metadata.name,
+                    onBackClick: () => { replicaSet && this._selectionActionCreator.selectItem({ item: replicaSet, itemUID: replicaSet.metadata.uid, selectedItemType: SelectedItemKeys.ReplicaSetKey, showSelectedItem: true }) }
```

Comment:
on backbutton, i am landing on the parent of the pod, which is replicaset etc, then if i do back of the history, then i am landing in unknown teritory, we should fix it as well.

Reply:
The forward back structure needs to be worked on throughout the UIs. I'll take that up separately